crates/storage/src/connection/transaction.rs

```diff
@@ -157,9 +177,21 @@ pub(super) fn transaction_with_receipt(
     let receipt: dto::Receipt =
         serde_json::from_slice(&receipt).context("Deserializing receipt")?;
 
+    let events = match row.get_ref_unwrap("events").as_blob_or_null()? {
+        Some(data) => data,
+        None => return Ok(None),
```

Comment:
It's probably possible for a transaction to have zero events - since we already return `None` if the receipt is missing, we should maybe just default to an empty events vector here?

Reply:
Not relevant - serde_json encodes empty vectors differently to null.